README.md

```diff
@@ -16,39 +16,137 @@ and to keep them as similar as possible:
+
+#### Privileged Mode
+
+In privileged mode the rpm-ostree compose uses container features
+itself, thus requires a privileged container in order to perform the
+compose. This is known as [recursive containers](https://github.com/projectatomic/bubblewrap/issues/284).
+
+### Container Build
 ---
 
-Here we store data in `/srv/coreos` on our host system.  You can choose
-any directory you like.  You should run these commands as `root`.
+To completely rebuild the coreos-assembler container image locally, execute
+`$ sudo podman build -t localhost/coreos-assembler .` or
+`$ sudo podman build -t localhost/coreos-assembler -f Dockerfile.rhel .`
+from the `coreos-assembler` repository. If building the RHEL version please 
+note you will need a maipo directory with references to
+proper repositories.
+
+You can optionally upload your built image to a registry such as quay.io by
+doing the following:
 
 ```
-$ mkdir /srv/coreos
-$ cd /srv/coreos
-$ alias coreos-assembler='podman run --rm -ti --privileged --userns=host -v $(pwd):/srv --workdir /srv quay.io/coreos-assembler/coreos-assembler'
+$ sudo podman push <image id> example.com/<account name>/coreos-assembler
 ```
 
-If you need access to CA certificates on your host (for example, when you need to access
-a git repo that is not on the public Internet), you can mount in the host certificates
-as read-only.  For example, on a Fedora host the alias would change to:
+### Setup
+---
 
-`$ alias coreos-assembler='podman run --rm -ti --privileged --userns=host -v /etc/pki:/etc/pki:ro -v $(pwd):/srv --workdir /srv quay.io/coreos-assembler/coreos-assembler'`
+Let's set up our working directory first. We'll create and use `./srv-coreos`
+on our host system. You can choose any directory you like. 
 
-See this [Stack Overflow question](https://stackoverflow.com/questions/26028971/docker-container-ssl-certificates) for additional discussion.
+```
+$ mkdir ./srv-coreos
+$ setfacl -m u:1000:rwx ./srv-coreos
+$ setfacl -d -m u:1000:rwx ./srv-coreos
+$ chcon system_u:object_r:container_file_t:s0 ./srv-coreos
+$ cd ./srv-coreos
+```
+
+In the above commands we:
+
+- created the `./srv-coreos` directory
+- set file ACLs so that the `builder` user (uid `1000` inside the container) can write files
+- set file ACLs so that new files get created with ACLs that allow the `builder` user
+- gave the directory an SELinux file context for sharing with containers
+- changed our working directory into `./srv-coreos`
+
+Now we'll define a bash function that we can use to call the assembler
+container:
+
+```
+$ coreos-assembler() {
+    env | grep COREOS_ASSEMBLER
+    set -x # so we can see what command gets run
+    sudo podman run --rm -ti -v ${PWD}:/srv/ --userns=host --device /dev/kvm --name coreos-assembler \
+               ${COREOS_ASSEMBLER_PRIVILEGED:+--privileged}                                          \
+               ${COREOS_ASSEMBLER_CONFIG_GIT:+-v $COREOS_ASSEMBLER_CONFIG_GIT:/srv/src/config/:ro}   \
```

Comment:
It'd be a bit nicer if we did something like auto-detect the case where this is a symlink and did a the bind mount over it or so.  But not necessary to do in this change.

Reply:
> It'd be a bit nicer if we did something like auto-detect the case where this is a symlink and did a the bind mount over it or so.

yeah, hadn't thought of that before. now that we're using a function and not an alias we could do that

> But not necessary to do in this change.

is that a +1 for merge?